yagmail/message.py

```diff
@@ -61,8 +62,8 @@ def prepare_message(
     # merge contents and attachments for now.
     if attachments is not None:
         for a in attachments:
-            if not isinstance(a, bytes) and not isinstance(a, tuple) and not os.path.isfile(a):
-                raise TypeError(f'{a} must be a valid filepath, bytes object or tuple (str, bytes). {a} is of type {type(a)}')
+            if not isinstance(a, io.IOBase) and not os.path.isfile(a):
+                raise TypeError(f'{a} must be a valid filepath or instance of io.IOBase. {a} is of type {type(a)}')
```

Comment:
Considering we should call it "file handler" as this might be more understandable, though less technically correct.

Reply:
How about `f'{a} must be a valid filepath or file handle (instance of io.IOBase). {a} is of type {type(a)}'` :)